Microsoft.Toolkit.Services/Services/MicrosoftGraph/MicrosoftGraphService.cs

```diff
@@ -186,22 +186,27 @@ public bool Initialize(string appClientId, ServicesToInitialize servicesToInitia
         /// Logout the current user
         /// </summary>
         /// <returns>success or failure</returns>
-        public virtual Task<bool> Logout()
+        public virtual async Task<bool> Logout()
         {
             if (!IsInitialized)
             {
                 throw new InvalidOperationException("Microsoft Graph not initialized.");
             }
 
-            IsAuthenticated = false;
             User = null;
 
+            bool result;
+
 #if WINRT
             var authenticationModel = AuthenticationModel.ToString();
-            return Authentication.LogoutAsync(authenticationModel);
+            result = await Authentication.LogoutAsync(authenticationModel);
 #else
-            return Task.Run(() => { return Authentication.Logout(); });
+            result = Authentication.Logout();
```

Comment:
I'm not sure how long this operation takes, but if it has the possibility to be long running (more than 100ms) then keep it as Task.Run as it already was

Reply:
I think that should be really fast, but this is for non-WINRT platform, which I don't have a device to validate, @nmetulev could you double confirm?

```c#
#if WINRT
            var authenticationModel = AuthenticationModel.ToString();
            result = await Authentication.LogoutAsync(authenticationModel);
#else
            result = Authentication.Logout();
#endif
```

Reply:
This operation is instantaneous, it only removes the user from the cache:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/src/Microsoft.Identity.Client/ClientApplicationBase.cs#L214

Reply:
And what if result = false? This might mean that the HttpRequest to the LogoutUrl didn't work, so is the user signed out or not? Does it make sense to set IsAuthenticated to false everytime?

Reply:
@azchohfi Good point, it's fixed now, this bug should already exist for a long time, but we indeed need to check the result.